x-pack/solutions/search/plugins/enterprise_search/server/routes/enterprise_search/search_applications.ts

```diff
@@ -131,6 +131,7 @@ export function registerSearchApplicationsRoutes({ log, router }: RouteDependenc
           name: request.params.engine_name,
           search_application: {
             indices: request.body.indices,
+            // @ts-expect-error elasticsearch@9.0.0 has this parameter been renamed to analytics_collection_name?
```

Comment:
This actually a different one. `name` is the search application's display name, `analytics_collection_name` is the name of the related collection
https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java#L67